src/params.ts

```diff
@@ -0,0 +1,14 @@
+export interface UrlParam<T> {
```

Comment:
I think we can place the type definition in `useParam.ts` since it's used there.